tripy/tripy/frontend/trace/ops/slice.py

```diff
@@ -30,17 +30,11 @@
 
 @dataclass(repr=False)
 class Slice(BaseTraceOp):
+    infer_rank = op_utils.InferRankPolicies.same_as_input()
```

Comment:
Maybe I'm missing something obvious, but is this actually correct? If the slice is by a single index, then the result has a lower rank than the input.

```
x = Tensor([1]) # rank 1
y = x[0] # rank 0
```

Reply:
Yes, correct, but we do the squeezing separately:
```py
    squeeze_dims = []
    for i, idx in enumerate(index):
        if isinstance(idx, (tuple, list)):
            raise NotImplementedError("Gather is not supported")
        if isinstance(idx, int):
            squeeze_dims.append(i)
    if squeeze_dims:
        out = squeeze(out, make_tuple(squeeze_dims))
```
The output of the `Slice` _trace_ operation, is always the same rank as the input, even though the frontend API can remove dimensions.